### PR TITLE
Added check to not fetch mercury prices when trigger capability is enabled

### PR DESCRIPTION
**Pull request by karen-stepanyan**
[DF-20169](https://smartcontract-it.atlassian.net/browse/DF-20169)

Added new check in `Observe` method in `core/services/relay/evm/mercury/v3/data_source.go` file that extracts the `EnableTriggerCapability` boolean from dataSource and if true skips calling `LatestPrice` function. For Data Feeds 2.0 with DON - DON communication we don't need to fetch any price info from mercury server

[DF-20169]: https://smartcontract-it.atlassian.net/browse/DF-20169?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by samsondav — DISMISSED** [review]

File: core/services/relay/evm/mercury/v3/data_source.go [hunk 1]
```diff
@@ -115,37 +124,43 @@ func (ds *datasource) Observe(ctx context.Context, repts ocrtypes.ReportTimestam
...
-				mercurytypes.PriceFeedMissingCount.WithLabelValues(ds.linkFeedID.String()).Inc()
-				ds.lggr.Warnw(fmt.Sprintf("Mercury server was missing LINK feed, using sentinel value of %s", v3.MissingPrice), "linkFeedID", ds.linkFeedID)
-				obs.LinkPrice.Val = v3.MissingPrice
-			} else if obs.LinkPrice.Err != nil {
-				mercurytypes.PriceFeedErrorCount.WithLabelValues(ds.linkFeedID.String()).Inc()
-				ds.lggr.Errorw("Mercury server returned error querying LINK price feed", "err", obs.LinkPrice.Err, "linkFeedID", ds.linkFeedID)
-			}
-		}()
+		// If the triggerCapability is enabled, we don't need to fetch the price
+		if !triggerCapabilityEnabled {
```
Comment (austinborn) [inline core/services/relay/evm/mercury/v3/data_source.go]: Could you condense this to `else if !triggerCapabilityEnabled` (same below)?

**Review by austinborn — APPROVED** [review]
Great, thank you

**Review by austinborn — DISMISSED** [review]

File: .changeset/wise-bears-end.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"chainlink": patch
+---
+
+#updated v3/data_source Observe method to skip calling LatestPrice when EnableTriggerCapability relay config is true
```
Comment (austinborn) [inline .changeset/wise-bears-end.md]:
```suggestion
---
"chainlink": minor
---

#updated mercury plugin to consider PluginConfig as optional if EnableTriggerCapability relay config is true. Then if PluginConfig is nil, skip fetching latestPrice for linkFeedId and nativeFeedId.
```

**Review by austinborn — APPROVED** [review]

**Review by samsondav — APPROVED** [review]
Bit of a hack, but then the entire plugin basically is one so 🤷‍♂️

**Comment by austinborn** [comment]
> Bit of a hack, but then the entire plugin basically is one so 🤷‍♂️

Thank you, since we're planning to move to the LLO plugin at some point I felt more comfortable taking this approach.

**Pull request merged**